src/libraries/System.IO.MemoryMappedFiles/tests/MemoryMappedFile.CreateFromFile.Tests.cs

```diff
@@ -868,5 +869,85 @@ public void ReusingNames_Windows(string name)
             }
         }
 
+        /// <summary>
+        /// Test that we can map special character devices on Unix using FileStream.
+        /// </summary>
+        [ConditionalTheory]
+        [InlineData(MemoryMappedFileAccess.Read)]
+        [InlineData(MemoryMappedFileAccess.ReadWrite)]
+        [InlineData(MemoryMappedFileAccess.Write)]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public void OpenCharacterDeviceAsStream(MemoryMappedFileAccess access)
+        {
+            string device = "/dev/zero";
+            if (!File.Exists(device))
+            {
+                throw new SkipTestException($"'{device}' is not available.");
+            }
+
+            long viewCapacity = 0xFF;
+
+            try
+            {
+                using (FileStream fs = new FileStream(device, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite))
+                using (MemoryMappedFile memMap = MemoryMappedFile.CreateFromFile(fs, null, viewCapacity, access, HandleInheritability.None, false))
+                using (MemoryMappedViewAccessor view = memMap.CreateViewAccessor(0, viewCapacity, access))
+                {
+                    if (access != MemoryMappedFileAccess.Write)
+                    {
+                        byte b = view.ReadByte(0);
+                        // /dev/zero return zeroes.
+                        Assert.Equal(0, b);
+                    }
+
+                    if (access != MemoryMappedFileAccess.Read)
+                    {
+                        view.Write(0, (byte)1);
+                    }
+                }
+            }
+            catch (UnauthorizedAccessException) { }
+            // ENODEV Operation not supported by device.
+            catch (IOException ex) when (ex.HResult == 19) { };
+        }
+
+        /// <summary>
+        /// Test that we can map special character devices on Unix using file name.
+        /// </summary>
+        [ConditionalTheory]
+        [InlineData(MemoryMappedFileAccess.Read)]
+        [InlineData(MemoryMappedFileAccess.ReadWrite)]
+        [PlatformSpecific(TestPlatforms.AnyUnix)]
+        public void OpenCharacterDeviceAsFile(MemoryMappedFileAccess access)
+        {
+            string device = "/dev/zero";
```

Comment:
Lines 923-929 has shared code. Any chance you could avoid repeating it? Maybe make the device and viewCapacity private members of the class, or wrap the repeated code with a shared private method.